feat(module:tabs): support destroy inactive lazy loaded tab when tabs change

Zz-ZzzZ opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: Closes #8591


## What is the new behavior?


## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

HyperLife1119 commented on components/tabs/tab-body.component.ts:
```diff
@@ -46,16 +46,29 @@ export class NzTabBodyComponent implements OnChanges {
     const { active } = changes;
-    if (active?.currentValue) {
-      this.hasBeenActive = true;
+    if (typeof active.currentValue === 'boolean') {
```
```suggestion
    if (typeof active?.currentValue === 'boolean') {
```

HyperLife1119 commented:
Can you update this demo?
https://ng.ant.design/components/tabs/zh#components-tabs-demo-lazy

Zz-ZzzZ commented:
> Can you update this demo? https://ng.ant.design/components/tabs/zh#components-tabs-demo-lazy

sure, do I need to add demo for nzDestroyInactiveTabPane?

HyperLife1119 commented:
You need to change this demo, the demo now needs to be combined with nzDestroyInactiveTabPane to work properly.

Zz-ZzzZ commented:
Sorry, there are some problems with the previous approach, which is my misunderstanding. the api added by this PR is supposed to destroy inactive lazy loaded tabs. 

Now I have changed the api name and added the demo usage instructions, please check

HyperLife1119 commented:
We need to keep the naming the same as antd, including the behavior, so you don't have to change the naming.

Zz-ZzzZ commented:
If the behavior needs to be fully synchronized with Antd, then I need to adjust the lazy-loaded policy, because in antd, destroyInactiveTabPane applies to all tab, so I should turn off this PR, which currently applies only to lazy-loaded tab

HyperLife1119 commented:
I'm closing this as #8845 provides a more accurate implementation :)
